packages/x-charts/src/themeAugmentation/components.d.ts

```diff
@@ -18,6 +18,10 @@ export interface ChartsComponents {
   MuiChartsClipPath?: {
     defaultProps?: ComponentsProps['MuiChartsClipPath'];
   };
+  MuiChartsGrid?: {
+    defaultProps?: ComponentsProps['MuiChartsGrid'];
+    styleOverrides?: ComponentsOverrides['MuiChartsGrid'];
```

Comment:
Only the `root` slot/class seems to work, because only that one is present on the `overridesResolver`.

Reply:
Nitpick, the `verticalLine` and `horizontalLine` slots will still not do anything. 🙈 
Consider applying the following diff to make them work:

```diff
diff --git a/packages/x-charts/src/ChartsGrid/ChartsGrid.tsx b/packages/x-charts/src/ChartsGrid/ChartsGrid.tsx
index f2d74db40..452083540 100644
--- a/packages/x-charts/src/ChartsGrid/ChartsGrid.tsx
+++ b/packages/x-charts/src/ChartsGrid/ChartsGrid.tsx
@@ -5,7 +5,11 @@ import { styled, useThemeProps } from '@mui/material/styles';
 
 import { CartesianContext } from '../context/CartesianContextProvider';
 import { useTicks } from '../hooks/useTicks';
-import { ChartsGridClasses, getChartsGridUtilityClass } from './chartsGridClasses';
+import {
+  ChartsGridClasses,
+  chartsGridClasses,
+  getChartsGridUtilityClass,
+} from './chartsGridClasses';
 
 const GridRoot = styled('g', {
   name: 'MuiChartsGrid',
@@ -16,7 +20,11 @@ const GridRoot = styled('g', {
 const GridLine = styled('line', {
   name: 'MuiChartsGrid',
   slot: 'Line',
-  overridesResolver: (props, styles) => styles.line,
+  overridesResolver: (props, styles) => [
+    { [`&.${chartsGridClasses.verticalLine}`]: styles.verticalLine },
+    { [`&.${chartsGridClasses.horizontalLine}`]: styles.horizontalLine },
+    styles.line,
+  ],
 })(({ theme }) => ({
   stroke: (theme.vars || theme).palette.divider,
   shapeRendering: 'crispEdges',
```

An example for reference:
https://github.com/mui/mui-x/blob/a3add0b4084f5a89466dade2f671d1339149dd1b/packages/x-date-pickers/src/DateTimePicker/DateTimePickerToolbar.tsx#L215-L218